elifepubmed/utils.py

```diff
@@ -130,3 +130,14 @@ def abstract_parts(abstract, label_types):
         if part:
             parts.append(part)
     return parts
+
+
+def first_property(the_object, properties):
+    """find the first non-None property of an object
+    
+    :param the_object: the object from which to get a property
+    :param properties: a list of property names to check if not None
+    """
+    for object_property in properties:
+        if hasattr(the_object, object_property) and getattr(the_object, object_property):
+            return getattr(the_object, object_property)
```

Comment:
This can also be written as:
`return next(getattr(object, p) for p in properties if hasattr(object, p))`

Reply:
Cool - which style do we prefer?

Reply:
I think it might need to be this though to check for empty values:

```
return next(getattr(object, p) for p in properties if hasattr(object, p) and getattr(object, p))
```

Reply:
Ah you're right, there was a mistake. Here's what you need:
```
In [1]: class Test:
   ...:     one = None
   ...:     two = 1
   ...:

In [2]: test = Test()

In [3]: next(getattr(test, p) for p in ['test', 'one', 'two'] if getattr(test, p, None))
Out[3]: 1
```